tasks/build.py

```diff
@@ -206,48 +318,145 @@ def apply_cvmfs_customizations(cvmfs_customizations, arch_job_dir):
 
-        download_pr(repo_name, branch_name, pr, arch_job_dir)
+    base_branch_name = pr.base.ref
+    log(f"{fn}(): pr.base.repo.ref '{base_branch_name}'")
 
-        # check if we need to apply local customizations:
-        #   is cvmfs_customizations defined? yes, apply it
-        apply_cvmfs_customizations(cvmfs_customizations, arch_job_dir)
-        # enlist jobs to proceed
-        job = Job(arch_job_dir, arch_target, slurm_opt)
-        jobs.append(job)
-    log("  %d jobs to proceed after applying white list" % len(jobs))
+    jobs = []
+    for arch, slurm_opt in arch_map.items():
+        # TODO check if arch in repos_map
+        arch_dir = arch.replace('/', '_')
+        # check if repo_target_map contains an entry for {arch}
+        if arch not in repocfg[REPO_TARGET_MAP]:
+            log(f"{fn}(): skipping arch {arch} because repo target map does not define repositories to build for")
+            continue
+        for repo_id in repocfg[REPO_TARGET_MAP][arch]:
+            # ensure repocfg contains information about the repository repo_id if repo_id != EESSI-pilot
+            # Note, EESSI-pilot is a bad/misleading name, it should be more like AS_IN_CONTAINER
+            if repo_id != "EESSI-pilot" and repo_id not in repocfg:
+                log(f"{fn}(): skipping repo {repo_id}, it is not defined in repo config {repocfg[REPOS_CFG_DIR]}")
+                continue
+            job_dir = os.path.join(event_dir, arch_dir, repo_id)
+            os.makedirs(job_dir, exist_ok=True)
+            log(f"{fn}(): job_dir '{job_dir}'")
+
+            # TODO optimisation? download once, copy and cleanup initial copy?
+            download_pr(base_repo_name, base_branch_name, pr, job_dir)
+
+            # prepare ./cfg/job.cfg
+            cpu_target = '/'.join(arch.split('/')[1:])
+            os_type = arch.split('/')[0]
+            prepare_job_cfg(job_dir, build_env_cfg, repocfg, repo_id, cpu_target, os_type)
+
+            # enlist jobs to proceed
+            job = Job(job_dir, arch, repo_id, slurm_opt)
+            jobs.append(job)
+
+    log(f"{fn}(): {len(jobs)} jobs to proceed after applying white list")
     if jobs:
         log(json.dumps(jobs, indent=4))
 
-    return repo_name, jobs
+    return jobs
+
+
+def prepare_job_cfg(job_dir, build_env_cfg, repos_cfg, repo_id, software_subdir, os_type):
+    """setup cfg/job.cfg
+
+    Args:
+        job_dir (string): directory of job
+        build_env_cfg (dictionary): build environment configuration
+        repos_cfg (dictionary):  configuration settings for all repositories
+        repo_id (string):  identifier of the repository to build for
+        software_subdir (string): software subdirectory to build for (CPU arch)
+        os_type (string): type of the os (e.g., linux)
+    """
+    fn = sys._getframe().f_code.co_name
+
+    jobcfg_dir = os.path.join(job_dir, 'cfg')
+    os.makedirs(jobcfg_dir, exist_ok=True)
```

Comment:
another place where we could use `try`/`except` around `makedirs`, which tells me we should have our own `mkdir` function (let's open an issue on that, it's just minor code cleanup)